### PR TITLE
Add ability to always reply when being limited

### PR DESCRIPTION
**Pull request by ak4zh**
Ability to send a rate-limited warning for every message received during the rate-limited time frame.

### CODE REVIEW

**Review by KnightNiwrem — APPROVED** [review]

File: src/typesAndDefaults.ts
Comment (KnorpelSenf) [inline src/typesAndDefaults.ts:51]:
```suggestion
   * @description Send the rate limited response for every message during the limit exceeded timeframe.
```

**Review by KnorpelSenf — APPROVED** [review]
LGTM, just a tiny nitpick about docs style

File: src/rateLimiter.ts
Comment (roj1512) [inline src/rateLimiter.ts:46]:
```suggestion
    if (hits === options.limit + 1 || (options.alwaysReply && hits > options.limit)) {
```

Comment (roj1512) [inline src/rateLimiter.ts:46]:
```suggestion
    if (hits === options.limit + 1 || (options.alwaysReply && hits > options.limit)) {
```

File: src/typesAndDefaults.ts
Comment (roj1512) [inline src/typesAndDefaults.ts:53]:
```suggestion
  alwaysReply?: boolean;
```

**Review by roj1512 — CHANGES_REQUESTED** [review]

**Review by roj1512 — CHANGES_REQUESTED** [review]
Please edit the PQ title to be more descriptive.

**Review by KnorpelSenf — APPROVED** [review]
I like the fact that it was renamed to `alwaysReply`, I wish I had suggested this myself. Thanks!

File: src/typesAndDefaults.ts
Comment (roj1512) [inline src/typesAndDefaults.ts:45]:
```suggestion
   * @description Executed only once for each limit excess unless alwaysReply is explicitly set to true. By default, it does nothing, meaning that the user is not notified when they exceed the limit. The middleware simply ignores excessive requests and the user is required to wait.
```

File: src/typesAndDefaults.ts
Comment (roj1512) [inline src/typesAndDefaults.ts:51]:
```suggestion
   * @description Whether to always call onLimitExceeded or not.
```

**Review by roj1512 — APPROVED** [review]
LGTM. Thank you so much!

**Comment by ak4zh** [comment]
Can you merge this change and push it to npm?

**Comment by roj1512** [comment]
@KnorpelSenf

**Comment by KnorpelSenf** [comment]
Sure, can do this later today. This plugin currently doesn't have a dedicated maintainer. @ak4zh are you interested in taking over?

**Comment by KnorpelSenf** [comment]
Then you would become a member of the grammyjs organisation on GitHub and you'll be able to merge and publish yourself :D

**Comment by ak4zh** [comment]
@KnorpelSenf Sounds interesting. I could try that to publish once I get approved.

**Comment by KnorpelSenf** [comment]
@ak4zh you now have an invitation to become a member of the grammY org. As soon as you accept, I'll make you the maintainer of this plugin, so you'll get admin rights for this repo.

Please DM me your npm name on Telegram (https://t.me/KnorpelSenf) so I can grant you publish permission for this package.

**Comment by KnorpelSenf** [comment]
@ak4zh are there any updates here?

**Comment by KnorpelSenf** [comment]
Are you going to perform a minor release?

**Comment by ak4zh** [comment]
@KnorpelSenf I have pushed the changes as a minor release, thanks for your guidance.

**Pull request merged**
